internal/hooks/hook_iterator.go

```diff
@@ -0,0 +1,52 @@
+package hooks
+
+import (
+	"github.com/launchdarkly/go-server-sdk/v7/ldhooks"
+)
+
+type hookIterator struct {
+	reverse    bool
+	cursor     int
+	collection *[]ldhooks.Hook
+}
+
+func newHookIterator(reverse bool, hooks *[]ldhooks.Hook) *hookIterator {
+	cursor := -1
```

Comment:
As a general comment, since the package is called `hooks` it would be implied that iterator/runner type names don't need the  "hook" prefix. 

For the iterator since it's private, probably doesn't matter. But for the runner, perhaps `hooks.Runner`  / `hooks.NewRunner()` would be cleaner.